### PR TITLE
MP-24 Implement accounts creation on Account Lookup Service (ALS)

### PR DESCRIPTION
**Pull request by kirgene**
* Implement POST /participants request flow logic
* Add test ALS server to be able to execute requests on a local machine.
* Add necessary OpenAPI description.
* Add test cases.
* Refactor code.

### CODE REVIEW

File: src/lib/model/AccountsModel.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+                return resolve();
+            });
+        });
+    }
+
+
+    /**
+     * Returns a promise that resolves when the state machine has reached a terminal state
+     */
+    async run() {
```
Comment (bushjames) [inline src/lib/model/AccountsModel.js]: Please check recent changes in the outbound model - I made a commit few days ago with updates for passing errors back to the backend.

File: src/index.js [hunk 2]
```diff
@@ -32,8 +33,8 @@ const { Logger, Transports } = require('@internal/log');
 
 const Cache = require('@internal/cache');
 
-const inboundApiSpec = yaml.load('./inboundApi/api.yaml');
-const outboundApiSpec = yaml.load('./outboundApi/api.yaml');
+const inboundApiSpec = yaml.load(fs.readFileSync('./inboundApi/api.yaml'));
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/index.js]: I changed yaml library from `yamljs` to `js-yaml`. If there are syntax errors in yaml, the old lib fails silently without any messages about what caused the error (I've spent some time figuring out why code fails).
Also, the new lib seems to be faster.

File: src/lib/model/AccountsModel.js [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+            this.data.response.push(...this._buildClientResponse(response));
+        }
+    }
+
+    _buildClientResponse(response) {
+        return response.partyList.map(party => ({
+            idType: party.partyId.partyIdType,
+            idValue: party.partyId.partyIdentifier,
+            ...!response.currency && {
+                error: {
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/lib/model/AccountsModel.js]: according to spec, if `currency` is not set in response, then changes were not applied on ALS server, so a user should know about it.

File: src/lib/model/AccountsModel.js (hunk 1 above)
Comment (kirgene) [reply src/lib/model/AccountsModel.js]: I've added necessary changes to the `catch` block.

File: src/lib/model/AccountsModel.js [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+            const requestKey = `${request.requestId}`;
+
+            this.subscriber.subscribe(requestKey);
+            this.subscriber.on('message', async (cn, msg) => {
+                try {
+                    let error;
+                    let message = JSON.parse(msg);
+
+                    if (message.type === 'accountsCreationErrorResponse') {
+                        error = new BackendError(`Got an error response creating accounts: ${util.inspect(message)}`, 500);
```
[the remark this answers is not in the document]
Comment (kirgene) [reply src/lib/model/AccountsModel.js]: done

File: src/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (bushjames) [reply src/index.js]: understood. that is OK.

File: src/lib/model/AccountsModel.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (bushjames) [reply src/lib/model/AccountsModel.js]: understood. thanks.

**Pull request merged**